packages/@react-spectrum/dialog/src/Dialog.tsx

```diff
@@ -104,6 +105,7 @@ function BaseDialog({children, slots, size, role, ...otherProps}: SpectrumBaseDi
           {[`spectrum-Dialog--${sizeVariant}`]: sizeVariant},
           otherProps.className
         )}
+        aria-labelledby={titleId}
```

Comment:
Can the Dialog be labelled using `aria-label` or and explicit `aria-labelledby` prop, without it being overwritten as `null`? I'm thinking of the DatePicker use case where the DialogTrigger supplies the label for the Dialog.

Reply:
I'm not sure about this, how would it work and when would we do one vs the other?
What if there is a DialogTrigger that supplies a labelled by AND there is a title in the dialog, do we want both?
I don't think this is currently interfering with aria-label, it's only forcing aria-labelledby

Reply:
explicitly setting `aria-labelledby` on the Dialog should take precedence.

Reply:
Also, `aria-labelledby` will take precedence over `aria-label`, so the internally set `aria-labelledby` referring to the slot, will override an explicitly set `aria-label` on the Dialog itself.